### ISSUE TITLE
HW optimized CRC32 implementation

### ISSUE THREAD

**Issue by pbadari** [awslabs/aws-checksums#82]
### Describe the feature

Looking through the code (for Intel architecture), I see that aws_checksums_crc32() is only SW implementation. There is no hardware accelerated implementation (SSE4.2 or AVX512). However aws_checksums_crc32c() has HW accelerated implementation.

Is there a reason why CRC32 implementation is not hardware accelerated? There are no users for it? 
Please let me know if there is interest.

### Use Case

HW optimized CRC32 implementation for Intel architecture using AVX512.

### Proposed Solution

I am working on AVX512 optimized CRC32c implementation (in AVX512 branch), I can easily modify it to support both CRC32 and CRC32c.

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by jmklix** [comment]
Yes, please make a PR if this is something that would be easy for you to add. We will make sure to review it.
